### ISSUE TITLE
Getting rid of SoLocator dialogue on ctrl-click

### ISSUE THREAD

**Issue by andreasneumann** [opengisch/solocator#37]
My colleagues used the SoLocator plugin a bit in the recent days.

We had some discussions today and came up with the idea of getting rid of the SoLocator Dialogue (the one that shows up on Ctrl-Click) altogether.

Reasoning:

- It makes loading layers quicker without that dialogue
- The ctrl-modifier would still be used to determine whether the layer should be loaded as Postgis or WMS
- The PostgreSQL-Servicename and Auth-Cfg could go into the SoLocator settings. They are typically not changed frequently
- Loading as separate WMS layers could go into the SoLocator settings
- Image Format (png|jpeg) can later be provided as a hint from the data product service, but until then, we'd like to have a white list of layers in the SoLocator plugin with data product ids that should be loaded as jpeg layers
- The choice of layers to load could be handled in our search service by exposing necessary childs of facade layers as regular single layers (and it is already improved a lot by commit 24581c1

The only thing we would really miss by removing the SoLocator Ctrl-Click dialogue is the description of a layer. We wonder, if the description could be displayed in a tooltip directly in the list of search results of SoLocator when the user selects an entry with a single-click? Note that the description of a layer requires a separate network request - it can only be retrieved by using the data product service.

![image](https://user-images.githubusercontent.com/884476/67214146-8429b480-f41f-11e9-9fda-53de3a673185.png)

Some of our other issues depend on this decision and could be closed or would have to be revised – so we need to clarify this first.

**Comment by 3nids** [comment]
I believe too that the dialog brings extra complexity which might not be required.

As a side note, one should remind that QGIS 3.6+ offers alternative actions for locator results (in the form of a menu accessible on right-click). We could have better tuning of the results there (eventually we could get the plugin to do the same in 3.4, but this has to be discussed).

The only solution I can think of is the following. Instead of showing a dialog, CTRL+click:
1. clears the results
2. Display Layer name as title with following actions:
  * Description on first line (no action) (self-note: https://doc.qt.io/qt-5/qtreeview.html#setFirstColumnSpanned)
  * Load as WMS
  * Load as PG
  * advanced options?…

Basically, this would move the dialog into the locator bar. This would look more integrated but in 3.4 we do not avoid the need for an extra click (except if we use SHIFT-click too or try to mimic the menu from QGIS 3.6+).

**Comment by andreasneumann** [comment]
We discussed this again - and we want to get rid of the dialogue alltogether. Most settings are available in the SoLocator settings alreay. What we still miss is the setting of
- choice of PG servicename
- possibility to select a PG auth config

The ctrl modifier key on double click should be used to distinguish between PG and WMS. There should be a SoLocator setting that allows to switch the default between PG and WMS or otherwise (internal vs external use of the plugin)
